### PR TITLE
Render widget groups with CSS grid

### LINKED ISSUE #1024: Fehler bei w50-Klasse
Bei der Verwendung der CSS-Klasse **w50** habe ich schon öfters das Problem gehabt, dass zum einen der Abstand der beidenden floatenden Container zur nächsten Überschrift darunter viel zu gering ist. Es ist quasi gar kein Abstand vorhanden. Zum anderen lässt sich das input-Feld der beiden w50-Container nicht anklicken. Ich kann nur über die TAB-Taste dahin gelangen und die Felder ausfüllen.

Das Problem habe momentan mit Metamodels, hatte es früher aber auch schon bei den Rocksolid Custom Elements. Daher könnte es vielleicht ein genereller Contao-Fehler sein?

Ich verwende Contao 4.8.5., der Fehler trat aber auch bei älteren Versionen schon auf. Getestet in sämtlichen Browsern auf dem Mac.

### LINKED ISSUE #1119: Neue CSS-Klassen für "full screen mode"
mit dem neuen "full screen mode" ist die kleinste Aufteilung der Eingabemasken vom 50% immer noch recht breit - die Idee von https://github.com/OMOSde/contao-om-backend/blob/master/FEATURES.md#17-backend-css finden wir vom "MM-Team" sehr gut...

Bitte kurze Rückmeldung, ob sowas in Planung ist oder aus OMO übernommen wird - ansonsten würden wir das für MM in unser CSS + Stylepicker einbauen.

### PR DESCRIPTION
**Pull request by m-vo**
| Q                | A
| -----------------| ---
| Fixed issues     | closes #1024, potentially closes #1119
| Docs PR or issue | todo

This is an early draft to render the BE widgets as grid elements. See https://github.com/contao/contao/pull/1824#issuecomment-645011607.

Open questions/notes:
* What's the browser support for the BE?
* What's our BC promise regarding CSS (classes) and BE markup?
* We might want to upgrade the CSS build chain first to support scss + autoprefixing before going further down this route...
* See https://bugs.chromium.org/p/chromium/issues/detail?id=375693 why we cannot use the `fieldset` as grid parent (`display:contents` + the definition on the parent instead would be another solution but this does not work on IE11).

### CODE REVIEW

**Comment by aschempp** [comment]
I honestly don't think CSS Grid is the right approach for widgets. The _overall layout_ should/could be Grid, but the widgets are flow elements that should use flex if anything. 

https://webdesign.tutsplus.com/articles/flexbox-vs-css-grid-which-should-you-use--cms-30184

> CSS Grid focuses on precise content placement. Each item is a grid cell, lined up along both a horizontal and a vertical axis. If you want to accurately control the position of items within a layout, CSS Grid is the way to go.

I don't think that's the case with the Contao back end DCA form.

**Comment by fritzmg** [comment]
You cannot do stuff like `.clr` with flex. But you can with grid.

**Comment by leofeyer** [comment]
Each single widget can be rendered with flex (if required), but they should be arranged on the page using grid.

**Comment by m-vo** [comment]
> I honestly don't think CSS Grid is the right approach for widgets. The overall layout should/could be Grid, but the widgets are flow elements that should use flex if anything.

Our DCA form is no different to a page layout: it has fixed-sized columns + you do not want to have reflow/relayout. It's a 2-dimensional problem (columns + aligned/"cleared" rows) which imo is best solved with grid.

Flexbox layouting is content-driven by nature. That's why it is so good for one-dimensional layouts without fixed sizes or for situations where you e.g. dynamically want to reflow elements into the next row and that take up all available space.

**Comment by zonky2** [comment]
the time is ready for grid
https://caniuse.com/css-grid

**Comment by m-vo** [comment]
Closing this as I currently do not have plans to continue this PR. Anyone interested feel free to pick up the work.

I think we should tackle these things as a part of a bigger BE rework when we know where we want to go. Otherwise we're likely breaking edge cases by implementing something new that wouldn't be future proof either.

File: core-bundle/src/Resources/contao/themes/flexible/basic.css [hunk 1]
```diff
@@ -162,10 +162,33 @@ p {
...
+.widget-group {
+	display: -ms-grid;
+	display: grid;
+	grid-column-gap: 30px;
+	-ms-grid-columns: 1fr 30px 1fr 30px 1fr 30px 1fr 30px 1fr 30px 1fr 30px 1fr 30px 1fr 30px 1fr 30px 1fr 30px 1fr 30px 1fr;
+	grid-template-columns: repeat(12, 1fr);
+}
+.widget-group > * {
+	-ms-grid-column: 1;
+	-ms-grid-column-span: 12;
```
Comment (ausi) [inline core-bundle/src/Resources/contao/themes/flexible/basic.css:177]: I think we should remove the `-ms-` prefixes ☺️

**Comment by m-vo** [comment]
Superseded by #7087.

**Pull request closed without merging**